### ISSUE TITLE
[BUG] PointerEvent float properties

### ISSUE THREAD

**Issue by rickyvetter** [ocaml-wasm/wasm_of_ocaml#105]
In practice all of the positional properties (`clientX/Y`, etc) on `pointerEvent`s are floats, not ints (at least in latest Chrome). In the `Dom_html` they inherit from `mouseEvent`, which has both as ints. In wasm this fails with a cast error (and in javacript obviously there is a loss of precision in the number).

I think the right fix is to update the types, ideally before the float/int breaking change in js_of_ocaml so that folks can migrate both at the same time. Maybe a closer audit of these values would be good to ensure we aren't missing more?

**Comment by vouillon** [comment]
Indeed, the type of these `MouseEvent` properties has been changed from int to float.
https://www.w3.org/TR/cssom-view-1/#extensions-to-the-mouseevent-interface

That would not be a bad thing to audit and modernize the Js_of_ocaml library.

**Comment by vouillon** [comment]
I have checked all integers properties.
See https://github.com/ocsigen/js_of_ocaml/pull/1747.
